cdap-app-fabric/src/main/java/io/cdap/cdap/app/guice/AppFabricServiceRuntimeModule.java

```diff
@@ -184,18 +185,25 @@
  */
 public final class AppFabricServiceRuntimeModule extends RuntimeModule {
 
+  public enum ServiceType {
+    SERVER,
+    PROCESSOR,
+  }
+
   public static final String NOAUTH_ARTIFACT_REPO = "noAuthArtifactRepo";
 
   private final CConfiguration cConf;
+  private final EnumSet<ServiceType> serviceType;
 
   @Inject
-  public AppFabricServiceRuntimeModule(CConfiguration cConf) {
+  public AppFabricServiceRuntimeModule(CConfiguration cConf, EnumSet<ServiceType> serviceType) {
```

Comment:
Generally in java, we use interfaces in param, i.e Set<> rather than concrete implementation.

Reply:
Done.